clients/githubrepo/contributors.go

```
@@ -42,50 +56,170 @@ func (handler *contributorsHandler) init(ctx context.Context, repourl *Repo) {
 	handler.contributors = nil
 }
 
-func (handler *contributorsHandler) setup() error {
+func (handler *contributorsHandler) setup(fileReader io.ReadCloser) error {
 	handler.once.Do(func() {
+		defer fileReader.Close()
```

what will this do if the repo doesn't have a CODEOWNERS file, so fileReader is nil?

I expected a panic, but not seeing one (don't have time to dig in right now).

---

also, should the `defer` go one line higher? otherwise the reader wont always get closed if we call this a second time for example.

---

Yea if the file reader is nil it will break let me fix that. I can also move the defer.

---

Re: nil codeowners file, the underlying type for our repo clients is an `*(os.File)`, which won't panic when `Close` is called, but rather return an error. So for now this seems to work, but may break in the future.